### ISSUE TITLE
\LimitIterator constructor is too restrictive

### ISSUE THREAD

**Issue by robations** [fusonic/linq#26]
It's possible to create a Linq object from a Traversable that can't be passed to `new \LimitIterator()` which accepts an `\Iterator`. Example:

```
<?php
class X implements IteratorAggregate
{
    public function getIterator()
    {
        return new ArrayIterator([]);
    }
}

$x = new X();
$y = Fusonic\Linq\Linq::from($x)
    ->take(5);

// Expected: $y instance of Linq
// Actual: PHP Catchable fatal error:  Argument 1 passed to LimitIterator::__construct()
//         must implement interface Iterator, instance of X given
```

From [this bug report](https://bugs.php.net/bug.php?id=52280), it seems like the solution is to use \IteratorIterator as a wrapper.

**Comment by davidroth** [comment]
Yep, using a wrapper `IteratorAggregate` would be ok. Though, we already have a hack to bypass another shortcoming of the `LimitIterator`:

https://github.com/fusonic/linq/blob/master/src/Fusonic/Linq/Linq.php#L116

Instead of adding another hack to fix a BCL issue, we should probably consider implementing a custom `LimitIterator`.

Nevertheless, for a quick fix the wrapper fix would be ok for me.
Are you working on a PR or is this up for grabs?

**Comment by robations** [comment]
I've got a PR in the works. From a glance at the code this might be an issue for `skip()` as well.
